src/liquid/components/ld-table/test/ld-table.spec.tsx

```diff
@@ -0,0 +1,553 @@
+              </ld-table-row>
+              <ld-table-row selectable>
+                <ld-table-cell>2</ld-table-cell>
+              </ld-table-row>
+              <ld-table-row selectable>
+                <ld-table-cell>3</ld-table-cell>
+              </ld-table-row>
+            </ld-table-body>
+          </ld-table>
+        ),
+      })
+
+      const getCheckboxes = () =>
+        Array.from(page.root.querySelectorAll('ld-table-row')).map((row) =>
+          row.shadowRoot.querySelector('ld-checkbox')
+        )
+      const checkboxes = getCheckboxes()
+
+      page.root.dispatchEvent(
+        new CustomEvent('ldTableSelectAll', {
+          bubbles: true,
+          detail: {
+            selected: true,
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      expect(checkboxes[1].checked).toBeTruthy()
+      expect(checkboxes[2].checked).toBeTruthy()
+      expect(checkboxes[3].checked).toBeTruthy()
+
+      page.root.dispatchEvent(
+        new CustomEvent('ldTableSelectAll', {
+          bubbles: true,
+          detail: {
+            selected: false,
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      expect(checkboxes[1].checked).toBeFalsy()
+      expect(checkboxes[2].checked).toBeFalsy()
+      expect(checkboxes[3].checked).toBeFalsy()
+
+      const prevented = new CustomEvent('ldTableSelectAll', {
+        bubbles: true,
+        detail: {
+          selected: true,
+        },
+      })
+      prevented.preventDefault()
+      page.root.dispatchEvent(prevented)
+
+      expect(checkboxes[1].checked).toBeFalsy()
+      expect(checkboxes[2].checked).toBeFalsy()
+      expect(checkboxes[3].checked).toBeFalsy()
+    })
+
+    it('updates select all checkbox on load', async () => {
+      const page = await newSpecPage({
+        components,
+        template: () => (
+          <ld-table>
+            <ld-table-head>
+              <ld-table-row selectable>
+                <ld-table-header>Nums</ld-table-header>
+              </ld-table-row>
+            </ld-table-head>
+            <ld-table-body>
+              <ld-table-row selectable>
+                <ld-table-cell>1</ld-table-cell>
+              </ld-table-row>
+              <ld-table-row selectable selected>
+                <ld-table-cell>2</ld-table-cell>
+              </ld-table-row>
+              <ld-table-row selectable>
+                <ld-table-cell>3</ld-table-cell>
+              </ld-table-row>
+            </ld-table-body>
+          </ld-table>
+        ),
+      })
+
+      await page.waitForChanges()
+      const getCheckboxes = () =>
+        Array.from(page.root.querySelectorAll('ld-table-row')).map((row) =>
+          row.shadowRoot.querySelector('ld-checkbox')
+        )
+      const checkboxes = getCheckboxes()
+
+      expect(checkboxes[0].indeterminate).toBeTruthy()
+    })
+
+    it('disabled select all if at least one row has a disabled selection', async () => {
```

Comment:
What if just one row is not selectable? Also, could the expected behavior also be, that all rows get selected, except for the disabled/non-selectable ones?

Reply:
This is more complex than it looks on first sight: If you click on the select all checkbox and one is not selectable you have to set the checkbox to either indeterminate, checked or unchecked, based on the state of all checkboxes. For this to work, you must disable the default behavior of the select all checkbox, and update it "manually". So far so good. But: Let's assume you clicked on the unchecked checkbox and it became indeterminate. Now you click on it again: What state should it have now? Checked would be the state that you would expect by intuition, but it must be unchecked, because it cannot be checked! That is why I decided to solve this problem by simply disabling this checkbox if there is a disabled row somewhere.